serverlessworkflow/modules/ROOT/pages/testing-and-troubleshooting/mocking-opnapi-services-with-wiremock.adoc

```diff
@@ -1 +1,294 @@
+<2> Defines a stub for the `/rest/v2/name/Greece` endpoint.
+<3> Defines the response for the stub.
+<4> Defines the response status.
+<5> Defines the response headers.
+<6> Defines the response body.
+--
+
+. Stop the WireMock server.
+
++
+--
+.Stopping a WireMock server example
+[source,java]
+----
+wireMockServer.stop();
+----
+--
+
+=== Starting a WireMock server as a `QuarkusTestResource`
+
+The following procedure describes how to start a WireMock server as a `QuarkusTestResource`.
+
+.Procedure
+
+. Create a class that implements the `QuarkusTestResource` interface.
+
++
+--
+.QuarkusTestResource implementation example
+[source,java]
+----
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+
+public class RestCountriesMockServer implements QuarkusTestResourceLifecycleManager {
+
+    private WireMockServer wireMockServer;
+
+    @Override
+    public Map<String, String> start() { <1>
+        configureWiremockServer();
+        return Map.of("quarkus.rest-client.restcountries_json.url", wireMockServer.baseUrl() + "/rest"); <2>
+    }
+
+    private void configureWiremockServer() {
+        wireMockServer = new WireMockServer(WireMockConfiguration.wireMockConfig().dynamicPort());
+        wireMockServer.start();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode greecePayload;
+        try {
+            greecePayload = objectMapper.readTree(getClass().getResourceAsStream("/country_mock.json"));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        wireMockServer.stubFor(get(urlEqualTo("/rest/v2/name/Greece"))
+                                       .willReturn(aResponse()
+                                                           .withStatus(200)
+                                                           .withHeader("Content-Type", "application/json")
+                                                           .withJsonBody(greecePayload)));
+    }
+
+    @Override
+    public void stop() { <3>
+        if (wireMockServer != null) {
+            wireMockServer.stop();
+        }
+    }
+}
+----
+<1> Start the test resource. This method is called once before all tests.
+<2> Returns a map of environment variables that will be set in the test environment. In this case, it sets the `quarkus.rest-client.restcountries_json.url` environment variable to the base URL of the WireMock server.
```

Comment:
```suggestion
<2> Returns a map of application properties that will be set in the test environment. In this case, it sets the `quarkus.rest-client.restcountries_json.url` property to the base URL of the WireMock server.
```

You can link the endpoint configuration guide here: https://github.com/kiegroup/kogito-docs/blob/main/serverlessworkflow/modules/ROOT/pages/service-orchestration/configuring-openapi-services-endpoints.adoc